ddtrace/profiling/collector/stack.pyx

```diff
@@ -244,7 +244,7 @@ cdef get_thread_name(thread_id):
             return "Anonymous Thread %d" % thread_id
 
 
-cdef stack_collect(ignore_profiler, thread_time, max_nframes, interval, wall_time, thread_span_links):
+cdef stack_collect(ignore_profiler, thread_time, max_nframes, interval, wall_time, thread_span_links) with gil:
```

Comment:
should we add a comment here on why? so it doesn't accidentally get removed in the future?

Reply:
Done :)
(I expect people to `git blame` but you never know :)

Reply:
yeah, it also helps if someone tries to copy/pasta the code without knowing why `with gil` was used too.